### PR TITLE
several packaging improvements

### PR DESCRIPTION
**Pull request by chipitsine**

### CODE REVIEW

**Comment by chipitsine** [comment]
not sure about bumping version

File: installer/tap-windows6.nsi [hunk 1]
```diff
@@ -271,6 +277,15 @@ Section -post
...
+	WriteRegStr HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "Publisher" "${PRODUCT_PUBLISHER}"
+	WriteRegStr HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "HelpLink" "https://openvpn.net/index.php/open-source.html"
+	WriteRegStr HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "URLInfoAbout" "https://openvpn.net"
+
+	${GetSize} "$INSTDIR" "/S=0K" $0 $1 $2
+	IntFmt $0 "0x%08X" $0
+	WriteRegDWORD HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "EstimatedSize" "$0"
 
 	; Advise a reboot
 	;Messagebox MB_OK "IMPORTANT: Rebooting the system is advised in order to finalize TAP driver installation/upgrade (this is an informational message only, pressing OK will not reboot)."
```
Comment (chipitsine) [inline installer/tap-windows6.nsi]: shall we also remove this commented Messagebox ?

Comment (mattock) [inline installer/tap-windows6.nsi]: The reboot advise was probably required in the old days (WinXP or even earlier). I have never rebooted after tap-windows6 install and have never had any issues. So I'd say remove the MessageBox.

File: installer/tap-windows6.nsi [hunk 2]
```diff
@@ -277,6 +277,15 @@ Section -post
...
 	WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "DisplayIcon" "$INSTDIR\icon.ico"
 	WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "DisplayVersion" "${PRODUCT_VERSION}"
+	WriteRegDWORD HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "NoModify" 1
+	WriteRegDWORD HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "NoRepair" 1
+	WriteRegStr HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "Publisher" "${PRODUCT_PUBLISHER}"
+	WriteRegStr HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "HelpLink" "https://openvpn.net/index.php/open-source.html"
+	WriteRegStr HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "URLInfoAbout" "https://openvpn.net"
+
+	${GetSize} "$INSTDIR" "/S=0K" $0 $1 $2
+	IntFmt $0 "0x%08X" $0
```
Comment (mattock) [inline installer/tap-windows6.nsi]: I could not find any reasonable explanation for IntFmt's format strings. What does this line put into $0?

Comment (chipitsine) [inline installer/tap-windows6.nsi]: same as printf(3)

**Comment by mattock** [comment]
I opened a [pull request](https://github.com/OpenVPN/tap-windows6/pull/25/) that adds build number to the tap-windows6 installer. While that is not directly related to this PR, it would be good to get it merged a.s.a.p. That would allow us to merge PRs like this one, which only touches .nsi files, and create an distinguishably named installer based on it.

**Comment by chipitsine** [comment]
can we merge it now?

**Comment by mattock** [comment]
@chipitsine : as mentioned as a line comment, makensis complains about an uninitialized variable ("PRODUCT_PUBLISHER"). Once that issue is fixed this PR is ready to be merged. You could add the variable to version.m4 or just remove the offending line in the .nsi file.

**Comment by chipitsine** [comment]
sorry, I did miss that line comment.

fixed, thanks

File: installer/tap-windows6.nsi [hunk 3]
```diff
@@ -271,9 +277,15 @@ Section -post
 	WriteRegExpandStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "UninstallString" "$INSTDIR\Uninstall.exe"
 	WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "DisplayIcon" "$INSTDIR\icon.ico"
 	WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "DisplayVersion" "${PRODUCT_VERSION}"
-
-	; Advise a reboot
-	;Messagebox MB_OK "IMPORTANT: Rebooting the system is advised in order to finalize TAP driver installation/upgrade (this is an informational message only, pressing OK will not reboot)."
+	WriteRegDWORD HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "NoModify" 1
+	WriteRegDWORD HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "NoRepair" 1
+	WriteRegStr HKLM "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT_NAME}" "Publisher" "${PRODUCT_PUBLISHER}"
```
Comment (chipitsine) [inline installer/tap-windows6.nsi]: NSIS estimates size during "folder choose" phase.
however, I do not know how to pick size from there.

current hack assumes that everthing reside in a folder and we calculate size of that folder.

Comment (chipitsine) [inline installer/tap-windows6.nsi]: size

**Pull request merged**